server_src/runners/state.js

```diff
@@ -3000,6 +2998,23 @@ class Runner {
         return false;
       });
   }
+  // Patch for updating OctoPrint's settings for now until re-work of printer cache with state.js.
+  static async getLatestOctoPrintSettingsValues(id) {
+    const index = _.findIndex(farmPrinters, function (o) {
+      return o._id == id;
+    });
+    // This is why the settings we're not updating! Forgot that connection options and preferences come in state, not settings/system.
+    await Runner.getState(id);
+    // Update the printers cached settings from OctoPrint
+    await Runner.getSettings(id);
+    // Update the printers cached system settings from OctoPrint
+    await Runner.getSystem(id);
+    // Re-generate the printer clean information - This is just cautionary, my tests showed it wasn't needed.
+    await PrinterClean.generate(
+      farmPrinters[index],
+      systemSettings.filamentManager
+    );
+  }
```

Comment:
New function for state. Just runs through the required calls to OctoPrint that are needed to update the printer settings on OctoFarms side. The extra generate wasn't needed but I'd rather keep it just encase there are timing issues server side.

Reply:
Can be cleaned up more once the refactor of state JS has happened.